madmom/ml/nn/__init__.py

```diff
@@ -92,6 +92,9 @@ def process(self, data, reset=True, **kwargs):
             Network predictions for this data.
 
         """
+        # make data at least 2d (required by NN-layers)
+        if data.ndim < 2:
+            data = np.array(data, subok=True, copy=False, ndmin=2)
```

Comment:
Yes, it is way faster on all machines I tested. I only spotted this when profiling the whole online beat tracking stuff, and it had a quite remarkable impact.